### PR TITLE
feat: implement create thread and run api

### PR DESCRIPTION
**Pull request by njhale**

### CODE REVIEW

**Comment by njhale** [comment]
I wrapped thread/message create with transactions.

**Review by thedadams — CHANGES_REQUESTED** [review]
It looks like the `ExtendedCreateThread` endpoint now has transactions for thread/message creation, but not the `ExtendedCreateAndRunThread`.

**Comment by njhale** [comment]
> It looks like the ExtendedCreateThread endpoint now has transactions for thread/message creation, but not the ExtendedCreateAndRunThread.

pushed the change. thanks for catching this.

File: pkg/server/extended.go [hunk 1]
```diff
@@ -1021,25 +1026,177 @@ func (s *Server) ExtendedCreateThread(w http.ResponseWriter, r *http.Request) {
...
 				w.WriteHeader(http.StatusBadRequest)
 				_, _ = w.Write([]byte(err.Error()))
-				return
+				return err
 			}
 		}
+
+		return nil
+	}); err != nil {
+		slog.Error("failed to create thread")
```
Comment (thedadams) [inline pkg/server/extended.go:1038]:
```suggestion
		slog.Error("failed to create thread")
		return
```

File: pkg/server/extended.go [hunk 2]
```diff
@@ -1021,25 +1026,177 @@ func (s *Server) ExtendedCreateThread(w http.ResponseWriter, r *http.Request) {
...
+		thread = new(db.Thread)
+	)
+	if err := gormDB.Transaction(func(tx *gorm.DB) error {
+		if err := create(tx, thread, publicThread); err != nil {
+			w.WriteHeader(http.StatusBadRequest)
+			_, _ = w.Write([]byte(err.Error()))
+			return err
+		}
+
+		if publicThread := createThreadAndRunRequest.Thread; publicThread != nil {
```
Comment (thedadams) [inline pkg/server/extended.go:1078]: Is it worth doing this?
```suggestion
		if publicThread := createThreadAndRunRequest.Thread; publicThread != nil && publicThread.Messages != nil {
```

**Review by thedadams — APPROVED** [review]
Looks good after `return` is added.

File: pkg/server/extended.go (hunk 2 above)
Comment (njhale) [reply pkg/server/extended.go]: I initially had this and removed it. Your comment has convinced me. Added.

File: pkg/server/extended.go (hunk 1 above)
Comment (njhale) [reply pkg/server/extended.go:1038]: Done

**Pull request merged**